tests/datasets/test_synthetic_lm_data.py

```diff
@@ -0,0 +1,117 @@
+from itertools import product
+
```

Comment:
Does it make sense to test whether the `[sep]`, `[mask]`, and `[cls]` tokens are present as expected? More generally, are there any assumptions that algorithms make about the structure of the input data, and do those need to be checked here?

Reply:
Yes! That is what `test_tokenizer_specific_properties` does!